lib/smart_name.rb

```diff
@@ -112,186 +125,22 @@ def == other
     other_key == key
   end
 
-  # ~~~~~~~~~~~~~~~~~~~ VARIANTS ~~~~~~~~~~~~~~~~~~~
-
-  def simple_key
-    decoded
-      .underscore
-      .gsub(/[^#{OK4KEY_RE}]+/, '_')
-      .split(/_+/)
-      .reject(&:empty?)
-      .map { |key| SmartName.stable_uninflect(key) }
-      .join('_')
-  end
-
-  def url_key
-    @url_key ||= part_names.map do |part_name|
-      stripped = part_name.decoded.gsub(/[^#{OK4KEY_RE}]+/, ' ').strip
-      stripped.gsub(/[\s\_]+/, '_')
-    end * self.class.joint
-  end
-
-  def safe_key
-    @safe_key ||= key.tr('*', 'X').tr self.class.joint, '-'
-  end
-
-  def decoded
-    @decoded ||= s.index('&') ? HTMLEntities.new.decode(s) : s
-  end
-
-  # ~~~~~~~~~~~~~~~~~~~ PARTS ~~~~~~~~~~~~~~~~~~~
-
-  alias simple? simple
-  def junction?
-    !simple?
```

Comment:
Now I'm starting to get it, a lot of it is must moving parts of it to modules, that's probably a good thing.